Missing Semigroup instance for Traversed

In upcoming `base-4.11`, `Semigroup` is a superclass of `Monoid`. Thus compiling `microlens-0.4.8.1` with ghc head results in following error:

```
--  While building package microlens-0.4.8.1 using:
      C:\Users\astro\AppData\Roaming\stack\setup-exe-cache\x86_64-windows\Cabal-simple_Z6RU0evB_2.1.0.0_ghc-8.5.20171205.exe --builddir=.stack-work\dist\50875de6 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: C:\Users\astro\haskell\asterius\.stack-work\logs\microlens-0.4.8.1.log

    Configuring microlens-0.4.8.1...
    Preprocessing library for microlens-0.4.8.1..
    Building library for microlens-0.4.8.1..
    [1 of 4] Compiling Lens.Micro.Type  ( src\Lens\Micro\Type.hs, .stack-work\dist\50875de6\build\Lens\Micro\Type.o )
    [2 of 4] Compiling Lens.Micro.Internal ( src\Lens\Micro\Internal.hs, .stack-work\dist\50875de6\build\Lens\Micro\Internal.o )
    [3 of 4] Compiling Lens.Micro       ( src\Lens\Micro.hs, .stack-work\dist\50875de6\build\Lens\Micro.o )

    C:\Users\astro\AppData\Local\Temp\stack9124\microlens-0.4.8.1\src\Lens\Micro.hs:1273:10: error:
        * Could not deduce (Semigroup (Traversed a f))
            arising from the superclasses of an instance declaration
          from the context: Applicative f
            bound by the instance declaration at src\Lens\Micro.hs:1273:10-48
        * In the instance declaration for `Monoid (Traversed a f)'
         |
    1273 | instance Applicative f => Monoid (Traversed a f) where
         |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/microlens-mtl/src/Lens/Micro/Mtl/Internal.hs
+++ b/microlens-mtl/src/Lens/Micro/Mtl/Internal.hs
@@ -180,10 +180,18 @@ newtype May a = May { getMay :: Maybe a }
 instance Monoid a => Monoid (May a) where
   mempty = May (Just mempty)
   {-# INLINE mempty #-}
+#if !MIN_VERSION_base(4,11,0)
   May Nothing `mappend` _ = May Nothing
   _ `mappend` May Nothing = May Nothing
   May (Just a) `mappend` May (Just b) = May (Just (mappend a b))
   {-# INLINE mappend #-}
+#else
+instance Semigroup a => Semigroup (May a) where
+  May Nothing <> _ = May Nothing
+  _ <> May Nothing = May Nothing
+  May (Just a) <> May (Just b) = May (Just (a <> b))
+  {-# INLINE (<>) #-}
+#endif
 
 ------------------------------------------------------------------------------
 -- FocusingMay
@@ -212,10 +220,18 @@ newtype Err e a = Err { getErr :: Either e a }
 instance Monoid a => Monoid (Err e a) where
   mempty = Err (Right mempty)
   {-# INLINE mempty #-}
+#if !MIN_VERSION_base(4,11,0)
   Err (Left e) `mappend` _ = Err (Left e)
   _ `mappend` Err (Left e) = Err (Left e)
   Err (Right a) `mappend` Err (Right b) = Err (Right (mappend a b))
   {-# INLINE mappend #-}
+#else
+instance Semigroup a => Semigroup (Err e a) where
+  Err (Left e) <> _ = Err (Left e)
+  _ <> Err (Left e) = Err (Left e)
+  Err (Right a) <> Err (Right b) = Err (Right (a <> b))
+  {-# INLINE (<>) #-}
+#endif
 
 ------------------------------------------------------------------------------
 -- FocusingErr
@@ -501,8 +517,14 @@ instance Functor (Effect m r) where
 instance (Monad m, Monoid r) => Monoid (Effect m r a) where
   mempty = Effect (return mempty)
   {-# INLINE mempty #-}
+#if !MIN_VERSION_base(4,11,0)
   Effect ma `mappend` Effect mb = Effect (liftM2 mappend ma mb)
   {-# INLINE mappend #-}
+#else
+instance (Monad m, Semigroup r) => Semigroup (Effect m r a) where
+  Effect ma <> Effect mb = Effect (liftM2 (<>) ma mb)
+  {-# INLINE (<>) #-}
+#endif
 
 instance (Monad m, Monoid r) => Applicative (Effect m r) where
   pure _ = Effect (return mempty)

--- a/microlens/src/Lens/Micro.hs
+++ b/microlens/src/Lens/Micro.hs
@@ -1273,8 +1273,14 @@ newtype Traversed a f = Traversed { getTraversed :: f a }
 instance Applicative f => Monoid (Traversed a f) where
   mempty = Traversed (pure (error "Lens.Micro.Traversed: value used"))
   {-# INLINE mempty #-}
+#if !MIN_VERSION_base(4,11,0)
   Traversed ma `mappend` Traversed mb = Traversed (ma *> mb)
   {-# INLINE mappend #-}
+#else
+instance Applicative f => Semigroup (Traversed a f) where
+  Traversed ma <> Traversed mb = Traversed (ma *> mb)
+  {-# INLINE (<>) #-}
+#endif
 
 newtype Bazaar a b t = Bazaar (forall f. Applicative f => (a -> f b) -> f t)
 

